### PR TITLE
refactor RibbonCommand to an interface and factory

### PR DESCRIPTION
**Pull request by spencergibb**
lays the groundworkd to implement an httpclient ribbon command

fixes gh-412

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
 		<eureka.version>1.1.147</eureka.version>
 		<feign.version>8.6.0</feign.version>
 		<hystrix.version>1.4.10</hystrix.version>
-		<ribbon.version>2.0.0</ribbon.version>
+		<ribbon.version>2.1.0</ribbon.version>
```
Comment (adriancole) [inline pom.xml]: hehe

Comment (spencergibb) [inline pom.xml]: yeah, I just haven't got around to finishing this PR up.

**Comment by spencergibb** [comment]
closed in favor of #498

**Comment by bogdatov** [comment]
Does that mean that SideCar now supports PATCH requests ?
If so which maven version should I use ?

**Comment by spencergibb** [comment]
@bogdatov this is a closed, out of date PR. See #498 which allows the _possibility_ to create a client that supports PATCH.

**Pull request closed without merging**